pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -118,6 +118,9 @@ func haveOverlap(a1, a2 []string) bool {
 }
 
 func (pl *VolumeRestrictions) PreFilter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod) (*framework.PreFilterResult, *framework.Status) {
+	if len(pod.Spec.Volumes) == 0 {
```

Comment:
Actually, even if an incoming pod has Volumes, we sometimes can return Skip. Please refer to `Filter` func.
https://github.com/kubernetes/kubernetes/blob/3aff1f97bef13f1d40e47fae785994d57c3e5a80/pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go#L175